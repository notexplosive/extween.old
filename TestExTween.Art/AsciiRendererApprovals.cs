using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using ExTween.Art;
using Xunit;

namespace TestExTween.Art
{
    [UseReporter(typeof(RiderReporter))]
    [UseApprovalSubdirectory("ApprovalTests")]
    public class AsciiRendererApprovals
    {
        [Fact]
        public void ascii_basic()
        {
            var painter = new AsciiPainter(new IntXyPair(10, 10));
            painter.DrawLine(new FloatXyPair(0, 0), new FloatXyPair(10, 10), 1, StrokeColor.Black);
            painter.DrawLine(new FloatXyPair(0, 10), new FloatXyPair(10, 0), 1, StrokeColor.Black);
            painter.DrawLine(new FloatXyPair(0, 5), new FloatXyPair(10, 5), 1, StrokeColor.Black);

            Approvals.Verify(painter.RenderString());
        }
    }
}
