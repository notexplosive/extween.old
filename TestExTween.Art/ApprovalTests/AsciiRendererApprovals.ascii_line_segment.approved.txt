                
                
                
                
                
   xxx          
      xxxx      
          x     
                
                
                
                
                
                
                
                

Record:
3, 5, x
3, 5, x
4, 5, x
5, 5, x
6, 6, x
7, 6, x
8, 6, x
9, 6, x
10, 7, x
